launcher/minecraft/mod/tasks/LocalResourcePackParseTask.cpp

```diff
@@ -178,6 +179,168 @@ bool processZIP(ResourcePack& pack, ProcessingLevel level)
     return true;
 }
 
+struct TextFormat {
+    QString color = "#000000";
+    bool bold = false;
+    bool italic = false;
+    bool underlined = false;
+    bool strikethrough = false;
+};
+
+bool textColorToHexColor(const QString& text_color, QString& result)
+{
+    static const QHash<QString, QString> text_to_hex = {
+        { "black", "#000000" },     { "dark_blue", "#0000AA" },    { "dark_green", "#00AA00" }, { "dark_aqua", "#00AAAA" },
+        { "dark_red", "#AA0000" },  { "dark_purple", "#AA00AA" },  { "gold", "#FFAA00" },       { "gray", "#AAAAAA" },
+        { "dark_gray", "#555555" }, { "blue", "#5555FF" },         { "green", "#55FF55" },      { "aqua", "#55FFFF" },
+        { "red", "#FF5555" },       { "light_purple", "#FF55FF" }, { "yellow", "#FFFF55" },     { "white", "#FFFFFF" },
+    };
+
+    auto it = text_to_hex.find(text_color);
+
+    if (it == text_to_hex.end()) {
+        qWarning() << "Invalid color string in component!";
+        result = "#000000"; // return black if color not found
+        return false;
+    }
+
+    result = it.value();
+    return true; 
+}
```

Comment:
I do believe HTML can handle the color names and you do not need to do any conversion.